packages/@extension-kid/notification-service/index.js

```diff
@@ -0,0 +1,40 @@
+import { ContainerModule } from "inversify";
```

Comment:
making this a `.ts`-file would remove the need for almost all type annotations. only `getExtensionModule` would still need its parameter to be typed to `extension: new (args: any[]) => {})` (at least this is what is inferred for inversify's `BindingToSyntax`-type.

Reply:
Package candidates in the `packages` directory technically may have `{main: "index.ts"}` as an entry point. Though as we extract them it is required to have `index.js` + `index.d.ts` combination. To avoid this confusion we decided to follow the `js` entry point from the get go.

Reply:
> Though as we extract them it is required to have `index.js` + `index.d.ts` combination.

Why exactly? We could tell webpack to grab those `.ts`-files anyway.

If we wanted to do things in a way that every js-user could use our stuff, shouldn't the other files be `.js` as well or is it only about the `main`?

Reply:
> Why exactly? We could tell webpack to grab and compile those `.ts`-files anyway.

That's what we do right now, yes, for packages in the `packages` directory. 
Though I'd like us to avoid a special config in webpack for anything that comes from npm. Including the packages we publish and own.

> shouldn't the other files be `.js` as well?

Totally! Those should eventually be compiled. It's just an attempt to balance the effort and confusion. As we extract those packages they basically stop working because the `main` script is `.ts`. While it's natural to setup the missing compilation step for the package, I discovered that it's very easy to forget about the `main` script

Reply:
> I'd like us to avoid a special config in webpack for anything that comes from npm
> ...
> I discovered that it's very easy to forget about the `main` script

Thanks, i really appreciate that you took the time to explain!  
Maybe we can have `.ts`-files as source as well as a dist in each package at some point in time, so we don't need to maintain the typings-files. But that's barely something to discuss now.


I'll mark this conversation as resolved.